Food security harvester

joaandrade opened the pull request:
Fixes #

### Features:
New harvester for the Food Security pilot outputs (VITO). This harvester allows to collect four new collections:

NEXTGEOSS_SENTINEL2_FAPAR (2017 - 2018)
NEXTGEOSS_SENTINEL2_FCOVER (2017 - 2018)
NEXTGEOSS_SENTINEL2_LAI (2017 - 2018)
NEXTGEOSS_SENTINEL2_NDVI (2017 - 2018)

Depending on the configuration, a different collection will be collected. So, 4 different harvesters must be created with the configurations below:

{
"start_date":"2017-01-01",
"collection":"FAPAR",
"username":"nextgeoss",
"password":"nextgeoss",
"make_private":false
}

or

{
"start_date":"2017-01-01",
"collection":"FCOVER",
"username":"nextgeoss",
"password":"nextgeoss",
"make_private":false
}

or

{
"start_date":"2017-01-01",
"collection":"LAI",
"username":"nextgeoss",
"password":"nextgeoss",
"make_private":false
}

or

{
"start_date":"2017-01-01",
"collection":"NDVI",
"username":"nextgeoss",
"password":"nextgeoss",
"make_private":false
}

VITO pilot (7.2) processed Data between 2017 and 2018. These harvesters (one per collection) are prepared to collect Data continously from a start date until the current date. The harvester is also prepare to continue if there is no Data for a specific date or time interval. You should configure the harvesters from 2017-01-01. Each job will harvest a maximum of 2 days of data (maximum of 120 datasets).

- [X] includes new  features
- [ ] includes tests covering changes
- [X] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes bugfix

Please [X] all the boxes above that apply

joaandrade commented:
In this case, the provider is VITO (same as Proba-V harvester). However, **the datasets / collections catalogued by this harvester must be related with the pilot 7.2 (Crop Monitoring Supporting Food Security).**

joaandrade commented:
The build is failing for a reason unrelated to this PR, it is failing in a Sentinel Harvester test.

joaandrade commented:
It makes sense that the first job only harvests a single dataset. The logic of the harvester is:

First JOB
1) Search by datasets between 2017-01-01 and 2017-01-02 (yyyy-mm-dd) -> No datasets available for these two days
2) Then search between 2017-01-02 and today (2019-05-23) -> Finds a huge number of datasets
3) Finds and saves the older date with datasets between the two dates above -> In this case it is 2017-02-15 is the first date with datasets (just one).
4) Harvests everything between (2017-02-15 and  2017-02-16) -> In this case is a single product from 2017-02-15

Second JOB
1) Find the date of the last product harvested -> It will be 2017-02-15 and use it as start date (end date will be +1 so 2017-02-16)
2) Finds the product that was already harvested and returns the message 's2a_20170215t105607z_29prr_fapar_10m_v001 will not be updated.'
3) Since the len(ids) == 0 because there are no products to harvest it will have the smae behavior of when there are no datasets for thos days.
4) Search between 2017-02-16 and today (2019-05-23) -> Finds a huge number of datasets
5) Finds and saves the older date with datasets between the two dates above -> In this case it is 2017-02-28 is the first date with datasets.
6) Harvests everything between (2017-02-28 and  2017-03-01) -> 81 products (so the second job should harvest 81 datasets)

And continues....

The problem in your case is that it seems the fetch are not cataloguing the first dataset. So, on the second job repeats the first job again. When the first dataset is on the catalogue, the second job will say that the first product was already catalogued and continues. On our side we harvest all the 2017 data.
FOR REFERENCE (starting at 2017-01-01): the first job will harvest only 1 dataset and the second job will harvest 81 datasets.

JGulic commented:
@joaandrade it's ok about the number of datasets. The problem is that the Harvesting process never ends.

We were able to find the error.  It's in gather_stage:
`ckan-dev_1    | 2019-05-23 11:47:47,761 INFO  [provider_logger] vito         | 2019-05-23 11:47:39.592011 | 200 | 1.88666s
ckan-dev_1    | 2019-05-23 11:47:51,561 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] next url: https://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&count=500&start=2017-01-02&end=2019-05-23&startIndex=501
ckan-dev_1    | Traceback (most recent call last):
ckan-dev_1    |   File "/usr/bin/paster", line 11, in <module>
ckan-dev_1    |     load_entry_point('PasteScript==2.0.2', 'console_scripts', 'paster')()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan-dev_1    |     invoke(command, command_name, options, args[1:])
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan-dev_1    |     exit_code = runner.run(args)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan-dev_1    |     result = self.command()
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/commands/harvester.py", line 185, in command
ckan-dev_1    |     gather_callback(consumer, method, header, body)
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 298, in gather_callback
ckan-dev_1    |     harvest_object_ids = gather_stage(harvester, job)
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 352, in gather_stage
ckan-dev_1    |     harvest_object_ids = harvester.gather_stage(job)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/ckanext/nextgeossharvest/harvesters/foodsecurity.py", line 231, in gather_stage
ckan-dev_1    |     restart_date = open_search_entry.find('dc:date').string.split('/')[1].split('T')[0]  # noqa: E501
ckan-dev_1    | AttributeError: 'NoneType' object has no attribute 'string'
ckan-dev_1    | 2019-05-23 11:47:52,054 INFO exited: harvest-gather-0 (exit status 1; not expected)
ckan-dev_1    | 2019-05-23 11:47:53,056 INFO spawned: 'harvest-gather-0' with pid 347
ckan-dev_1    | 2019-05-23 11:47:54,057 INFO success: harvest-gather-0 entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)`


Full logs:
`ckan-dev_1    | ----------------------------------------2019-05-23 11:47:38,488 INFO  [ckanext.harvest.logic.action.create] Harvest job create: {'source_id': u'a0191215-db3a-418a-807c-3709bd32cb61', 'run': True}
ckan-dev_1    | 2019-05-23 11:47:38,506 INFO  [ckanext.harvest.logic.action.create] Harvest job saved 1d7d6bd4-4e63-4243-8c3c-92db5317676d
ckan-dev_1    | 2019-05-23 11:47:38,507 INFO  [ckanext.harvest.logic.action.update] Send job to gather queue: {'id': u'1d7d6bd4-4e63-4243-8c3c-92db5317676d'}
solr          | 378368 INFO  (qtp1989972246-22) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=name:"a0191215-db3a-418a-807c-3709bd32cb61"+OR+id:"a0191215-db3a-418a-807c-3709bd32cb61"&fq=site_id:"default"&rows=1&wt=json} hits=1 status=0 QTime=0
ckan-dev_1    | 2019-05-23 11:47:38,577 INFO  [ckanext.harvest.logic.action.update] Sent job 1d7d6bd4-4e63-4243-8c3c-92db5317676d to the gather queue
ckan-dev_1    | 2019-05-23 11:47:38,577 DEBUG [ckanext.harvest.queue] Received harvest job id: 1d7d6bd4-4e63-4243-8c3c-92db5317676d
ckan-dev_1    | 2019-05-23 11:47:38,582 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'username': u'nextgeoss', u'make_private': False, u'password': u'nextgeoss', u'start_date': u'2017-01-01', u'collection': u'FAPAR'}
ckan-dev_1    | 2019-05-23 11:47:38,583 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] Food security harvester gather_stage for job: <HarvestJob id=1d7d6bd4-4e63-4243-8c3c-92db5317676d created=2019-05-23 11:47:38.503296 gather_started=2019-05-23 11:47:38.582919 gather_finished=None finished=None source_id=a0191215-db3a-418a-807c-3709bd32cb61 status=Running>
ckan-dev_1    | 2019-05-23 11:47:38,590 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] Harvesting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-01&end=2017-01-02&count=500
ckan-dev_1    | 2019-05-23 11:47:38,590 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] getting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-01&end=2017-01-02&count=500
ckan-dev_1    | 2019-05-23 11:47:38,749 INFO  [provider_logger] vito         | 2019-05-23 11:47:38.590960 | 200 | 0.156237s
ckan-dev_1    | 2019-05-23 11:47:38,751 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] next url: None
solr          | 378654 INFO  (qtp1989972246-15) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=name:"a0191215-db3a-418a-807c-3709bd32cb61"+OR+id:"a0191215-db3a-418a-807c-3709bd32cb61"&fq=site_id:"default"&rows=1&wt=json} hits=1 status=0 QTime=0
solr          | 378837 INFO  (qtp1989972246-17) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&facet.limit=50&fl=id+validated_data_dict&sort=score+desc,+metadata_modified+desc&fq=%2Bcapacity:public+harvest_source_id:"a0191215-db3a-418a-807c-3709bd32cb61"+-dataset_type:harvest&fq=%2Bsite_id:"default"&fq=%2Bstate:active&facet.mincount=1&rows=11&facet=true&wt=json} hits=0 status=0 QTime=0
solr          | 378871 INFO  (qtp1989972246-18) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&facet.limit=50&fl=id+validated_data_dict&sort=score+desc,+metadata_modified+desc&fq=%2Bcapacity:public+harvest_source_id:"a0191215-db3a-418a-807c-3709bd32cb61"+-dataset_type:harvest&fq=%2Bsite_id:"default"&fq=%2Bstate:active&facet.mincount=1&rows=11&facet=true&wt=json} hits=0 status=0 QTime=0
ckan-dev_1    | 2019-05-23 11:47:39,592 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] getting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-02&end=2019-05-23&count=500
solr          | 383399 INFO  (qtp1989972246-20) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=name:"a0191215-db3a-418a-807c-3709bd32cb61"+OR+id:"a0191215-db3a-418a-807c-3709bd32cb61"&fq=site_id:"default"&rows=1&wt=json} hits=1 status=0 QTime=0
solr          | 383514 INFO  (qtp1989972246-17) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&facet.limit=50&fl=id+validated_data_dict&sort=score+desc,+metadata_modified+desc&fq=%2Bcapacity:public+harvest_source_id:"a0191215-db3a-418a-807c-3709bd32cb61"+-dataset_type:harvest&fq=%2Bsite_id:"default"&fq=%2Bstate:active&facet.mincount=1&rows=11&facet=true&wt=json} hits=0 status=0 QTime=0
solr          | 383525 INFO  (qtp1989972246-18) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&facet.limit=50&fl=id+validated_data_dict&sort=score+desc,+metadata_modified+desc&fq=%2Bcapacity:public+harvest_source_id:"a0191215-db3a-418a-807c-3709bd32cb61"+-dataset_type:harvest&fq=%2Bsite_id:"default"&fq=%2Bstate:active&facet.mincount=1&rows=11&facet=true&wt=json} hits=0 status=0 QTime=0
ckan-dev_1    | 2019-05-23 11:47:47,761 INFO  [provider_logger] vito         | 2019-05-23 11:47:39.592011 | 200 | 1.88666s
ckan-dev_1    | 2019-05-23 11:47:51,561 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] next url: https://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&count=500&start=2017-01-02&end=2019-05-23&startIndex=501
ckan-dev_1    | Traceback (most recent call last):
ckan-dev_1    |   File "/usr/bin/paster", line 11, in <module>
ckan-dev_1    |     load_entry_point('PasteScript==2.0.2', 'console_scripts', 'paster')()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan-dev_1    |     invoke(command, command_name, options, args[1:])
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan-dev_1    |     exit_code = runner.run(args)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan-dev_1    |     result = self.command()
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/commands/harvester.py", line 185, in command
ckan-dev_1    |     gather_callback(consumer, method, header, body)
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 298, in gather_callback
ckan-dev_1    |     harvest_object_ids = gather_stage(harvester, job)
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 352, in gather_stage
ckan-dev_1    |     harvest_object_ids = harvester.gather_stage(job)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/ckanext/nextgeossharvest/harvesters/foodsecurity.py", line 231, in gather_stage
ckan-dev_1    |     restart_date = open_search_entry.find('dc:date').string.split('/')[1].split('T')[0]  # noqa: E501
ckan-dev_1    | AttributeError: 'NoneType' object has no attribute 'string'
ckan-dev_1    | 2019-05-23 11:47:52,054 INFO exited: harvest-gather-0 (exit status 1; not expected)
ckan-dev_1    | 2019-05-23 11:47:53,056 INFO spawned: 'harvest-gather-0' with pid 347
ckan-dev_1    | 2019-05-23 11:47:54,057 INFO success: harvest-gather-0 entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
solr          | 394399 INFO  (qtp1989972246-20) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&rows=1&wt=json} hits=2427 status=0 QTime=0
solr          | 394408 INFO  (qtp1989972246-21) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/admin/file params={file=schema.xml} status=0 QTime=0
solr          | 394992 INFO  (qtp1989972246-15) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&rows=1&wt=json} hits=2427 status=0 QTime=0
solr          | 394997 INFO  (qtp1989972246-20) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/admin/file params={file=schema.xml} status=0 QTime=0
ckan-dev_1    | 2019-05-23 11:47:55,189 DEBUG [ckanext.spatial.plugin] Setting up the spatial model
ckan-dev_1    | 2019-05-23 11:47:55,201 DEBUG [ckanext.spatial.model.package_extent] Spatial tables defined in memory
ckan-dev_1    | 2019-05-23 11:47:55,207 DEBUG [ckanext.spatial.model.package_extent] Spatial tables already exist
ckan-dev_1    | 2019-05-23 11:47:55,231 DEBUG [ckanext.harvest.model] Harvest tables defined in memory
ckan-dev_1    | 2019-05-23 11:47:55,233 DEBUG [ckanext.harvest.model] Harvest tables already exist`

davidcordeiro-deimos commented:
@JGulic we have updated the code to:

open_search_entry protected against no entries;
metadata url check updated from 'HMA' to 'HMA ordering'; (due to change in VITO interface)

We've tested and we were able to harvest data starting on 2017-01-01. The harvester should run continuously until it reaches 2019. After 2019 all the jobs won't harvest datasets because there are no more data, but the harvester will keep running.

georgiana-b commented:
Hello @joaandrade!

I just tried harvesting `FAPAR` with the latest code. It ran for about 2 hours but didn't collect anything even though it was following the steps you described [above](https://github.com/NextGeoss/ckanext-nextgeossharvest/pull/90#issuecomment-495185037).

There were also 2 errors (the second one might have been caused by the first):

```
ckan-dev_1    | 2019-05-27 10:22:09,957 INFO  [ckanext.harvest.logic.action.create] Harvest job create: {'source_id': u'a0191215-db3a-418a-807c-3709bd32cb61', 'run': True}
ckan-dev_1    | 2019-05-27 10:22:09,979 INFO  [ckanext.harvest.logic.action.create] Harvest job saved 0c2900e0-77e3-4093-be71-bd273ee22245
ckan-dev_1    | 2019-05-27 10:22:09,979 INFO  [ckanext.harvest.logic.action.update] Send job to gather queue: {'id': u'0c2900e0-77e3-4093-be71-bd273ee22245'}
solr          | 241941 INFO  (qtp1989972246-21) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=name:"a0191215-db3a-418a-807c-3709bd32cb61"+OR+id:"a0191215-db3a-418a-807c-3709bd32cb61"&fq=site_id:"default"&rows=1&wt=json} hits=1 status=0 QTime=0
ckan-dev_1    | 2019-05-27 10:22:10,058 INFO  [ckanext.harvest.logic.action.update] Sent job 0c2900e0-77e3-4093-be71-bd273ee22245 to the gather queue
ckan-dev_1    | 2019-05-27 10:22:10,058 DEBUG [ckanext.harvest.queue] Received harvest job id: 0c2900e0-77e3-4093-be71-bd273ee22245
ckan-dev_1    | 2019-05-27 10:22:10,064 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'username': u'nextgeoss', u'make_private': False, u'password': u'nextgeoss', u'start_date': u'2017-01-01', u'collection': u'FAPAR'}
ckan-dev_1    | 2019-05-27 10:22:10,064 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] Food security harvester gather_stage for job: <HarvestJob id=0c2900e0-77e3-4093-be71-bd273ee22245 created=2019-05-27 10:22:09.975575 gather_started=2019-05-27 10:22:10.064215 gather_finished=None finished=None source_id=a0191215-db3a-418a-807c-3709bd32cb61 status=Running>
ckan-dev_1    | 2019-05-27 10:22:10,073 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] Harvesting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-01&end=2017-01-02&count=500
ckan-dev_1    | 2019-05-27 10:22:10,073 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] getting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-01&end=2017-01-02&count=500
ckan-dev_1    | 2019-05-27 10:22:10,233 INFO  [provider_logger] vito         | 2019-05-27 10:22:10.073315 | 200 | 0.157576s
ckan-dev_1    | 2019-05-27 10:22:10,235 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] next url: None
solr          | 242348 INFO  (qtp1989972246-22) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=name:"a0191215-db3a-418a-807c-3709bd32cb61"+OR+id:"a0191215-db3a-418a-807c-3709bd32cb61"&fq=site_id:"default"&rows=1&wt=json} hits=1 status=0 QTime=0
solr          | 242491 INFO  (qtp1989972246-13) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&facet.limit=50&fl=id+validated_data_dict&sort=score+desc,+metadata_modified+desc&fq=%2Bcapacity:public+harvest_source_id:"a0191215-db3a-418a-807c-3709bd32cb61"+-dataset_type:harvest&fq=%2Bsite_id:"default"&fq=%2Bstate:active&facet.mincount=1&rows=11&facet=true&wt=json} hits=0 status=0 QTime=0
solr          | 242503 INFO  (qtp1989972246-18) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&facet.limit=50&fl=id+validated_data_dict&sort=score+desc,+metadata_modified+desc&fq=%2Bcapacity:public+harvest_source_id:"a0191215-db3a-418a-807c-3709bd32cb61"+-dataset_type:harvest&fq=%2Bsite_id:"default"&fq=%2Bstate:active&facet.mincount=1&rows=11&facet=true&wt=json} hits=0 status=0 QTime=0
ckan-dev_1    | 2019-05-27 10:22:11,074 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] getting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-02&end=2019-05-27&count=500
ckan-dev_1    | 2019-05-27 10:22:21,624 INFO  [provider_logger] vito         | 2019-05-27 10:22:11.074446 | 200 | 2.335006s
ckan-dev_1    | 2019-05-27 10:22:24,985 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] next url: https://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&count=500&start=2017-01-02&end=2019-05-27&startIndex=501
ckan-dev_1    | Traceback (most recent call last):
ckan-dev_1    |   File "/usr/bin/paster", line 11, in <module>
ckan-dev_1    |     load_entry_point('PasteScript==2.0.2', 'console_scripts', 'paster')()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan-dev_1    |     invoke(command, command_name, options, args[1:])
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan-dev_1    |     exit_code = runner.run(args)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan-dev_1    |     result = self.command()
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/commands/harvester.py", line 185, in command
ckan-dev_1    |     gather_callback(consumer, method, header, body)
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 298, in gather_callback
ckan-dev_1    |     harvest_object_ids = gather_stage(harvester, job)
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 352, in gather_stage
ckan-dev_1    |     harvest_object_ids = harvester.gather_stage(job)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/ckanext/nextgeossharvest/harvesters/foodsecurity.py", line 231, in gather_stage
ckan-dev_1    |     restart_date = open_search_entry.find('dc:date').string.split('/')[1].split('T')[0]  # noqa: E501
ckan-dev_1    | AttributeError: 'NoneType' object has no attribute 'string'
```

Followed by an indexing error:

```
ckan-dev_1    | 2019-05-27 11:07:39,993 ERROR [ckan.lib.search] Error while indexing dataset ddb01785-601e-4534-8d45-cb21a0e0b9e1: ValueError("Expecting ',' delimiter: line 1 column 49 (char 48)",)
ckan-dev_1    | Traceback (most recent call last):
ckan-dev_1    |   File "/usr/bin/paster", line 11, in <module>
ckan-dev_1    |     load_entry_point('PasteScript==2.0.2', 'console_scripts', 'paster')()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan-dev_1    |     invoke(command, command_name, options, args[1:])
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan-dev_1    |     exit_code = runner.run(args)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan-dev_1    |     result = self.command()
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/cli.py", line 554, in command
ckan-dev_1    |     self.rebuild()
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/cli.py", line 577, in rebuild
ckan-dev_1    |     quiet=self.options.quiet)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/__init__.py", line 194, in rebuild
ckan-dev_1    |     defer_commit
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/index.py", line 101, in update_dict
ckan-dev_1    |     self.index_package(pkg_dict, defer_commit)
ckan-dev_1    |   File "/srv/app/src/ckan/ckan/lib/search/index.py", line 278, in index_package
ckan-dev_1    |     pkg_dict = item.before_index(pkg_dict)
ckan-dev_1    |   File "/srv/app/src_extensions/ckanext-nextgeoss/ckanext/nextgeoss/plugin.py", line 231, in before_index
ckan-dev_1    |     geometry = json.loads(geometry)
ckan-dev_1    |   File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
ckan-dev_1    |     return _default_decoder.decode(s)
ckan-dev_1    |   File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
ckan-dev_1    |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
ckan-dev_1    |   File "/usr/lib/python2.7/json/decoder.py", line 380, in raw_decode
ckan-dev_1    |     obj, end = self.scan_once(s, idx)
ckan-dev_1    | ValueError: Expecting ',' delimiter: line 1 column 49 (char 48)
```

I tried to run it a second time and got the same error.
This is the config I was using:

```json
{
  "start_date":"2017-01-01",
  "collection":"FAPAR",
  "username":"nextgeoss",
  "password":"nextgeoss",
  "make_private":false
}
```
Please look into it and let me know if you have trouble reproducing the error or if there's anything else I can help with.

joaandrade commented:
Hi @georgiana-b,

I think you are not testing the latest version of the code. In your logs the error appears on line 231:
restart_date = open_search_entry.find('dc:date').string.split('/')[1].split('T')[0]  # noqa: E501

In the latest version of the code, this line is not on the line 231 anymore. This was the piece of code we've updated. Can you please confirm that you are not using the latest version?

Regards,
João

georgiana-b commented:
Might have been the docker image cache. Let me try that again and get back to you.

georgiana-b commented:
@joaandrade You were right that I wasn't using the latest code. So I updated the code and ran it again. Unfortunately I get the same error in the latest version as well, just on a different line:

```
ckan-dev_1    | 2019-05-27 12:56:05,454 INFO  [ckanext.harvest.logic.action.create] Harvest job create: {'source_id': u'0683bd8b-673b-4fc9-98fd-1f153e1e3144', 'run': True}
ckan-dev_1    | 2019-05-27 12:56:05,488 INFO  [ckanext.harvest.logic.action.create] Harvest job saved 9f469416-292e-46f7-97d4-44250a27e16f
ckan-dev_1    | 2019-05-27 12:56:05,489 INFO  [ckanext.harvest.logic.action.update] Send job to gather queue: {'id': u'9f469416-292e-46f7-97d4-44250a27e16f'}
solr          | 205988 INFO  (qtp1989972246-21) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=name:"0683bd8b-673b-4fc9-98fd-1f153e1e3144"+OR+id:"0683bd8b-673b-4fc9-98fd-1f153e1e3144"&fq=site_id:"default"&rows=1&wt=json} hits=0 status=0 QTime=8
ckan-dev_1    | 2019-05-27 12:56:05,598 INFO  [ckanext.harvest.logic.action.update] Sent job 9f469416-292e-46f7-97d4-44250a27e16f to the gather queue
ckan-dev_1    | 2019-05-27 12:56:05,598 DEBUG [ckanext.harvest.queue] Received harvest job id: 9f469416-292e-46f7-97d4-44250a27e16f
ckan-dev_1    | 2019-05-27 12:56:05,604 DEBUG [ckanext.nextgeossharvest.lib.nextgeoss_base] Using config: {u'username': u'nextgeoss', u'make_private': False, u'password': u'nextgeoss', u'start_date': u'2017-01-01', u'collection': u'FAPAR'}
ckan-dev_1    | 2019-05-27 12:56:05,604 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] Food security harvester gather_stage for job: <HarvestJob id=9f469416-292e-46f7-97d4-44250a27e16f created=2019-05-27 12:56:05.485159 gather_started=2019-05-27 12:56:05.604164 gather_finished=None finished=None source_id=0683bd8b-673b-4fc9-98fd-1f153e1e3144 status=Running>
ckan-dev_1    | 2019-05-27 12:56:05,612 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] Harvesting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-01&end=2017-01-02&count=500
ckan-dev_1    | 2019-05-27 12:56:05,613 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] getting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-01&end=2017-01-02&count=500
solr          | 206385 INFO  (qtp1989972246-17) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=name:"0683bd8b-673b-4fc9-98fd-1f153e1e3144"+OR+id:"0683bd8b-673b-4fc9-98fd-1f153e1e3144"&fq=site_id:"default"&rows=1&wt=json} hits=0 status=0 QTime=0
solr          | 206728 INFO  (qtp1989972246-20) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&facet.limit=50&fl=id+validated_data_dict&sort=score+desc,+metadata_modified+desc&fq=%2Bcapacity:public+harvest_source_id:"0683bd8b-673b-4fc9-98fd-1f153e1e3144"+-dataset_type:harvest&fq=%2Bsite_id:"default"&fq=%2Bstate:active&facet.mincount=1&rows=11&facet=true&wt=json} hits=0 status=0 QTime=11
ckan-dev_1    | 2019-05-27 12:56:06,278 INFO  [provider_logger] vito         | 2019-05-27 12:56:05.613075 | 200 | 0.661981s
ckan-dev_1    | 2019-05-27 12:56:06,280 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] next url: None
solr          | 206760 INFO  (qtp1989972246-19) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&facet.limit=50&fl=id+validated_data_dict&sort=score+desc,+metadata_modified+desc&fq=%2Bcapacity:public+harvest_source_id:"0683bd8b-673b-4fc9-98fd-1f153e1e3144"+-dataset_type:harvest&fq=%2Bsite_id:"default"&fq=%2Bstate:active&facet.mincount=1&rows=11&facet=true&wt=json} hits=0 status=0 QTime=1
ckan-dev_1    | 2019-05-27 12:56:06,613 INFO  [ckanext.nextgeossharvest.harvesters.foodsecurity] getting http://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&start=2017-01-02&end=2019-05-27&count=500
ckan-dev_1    | Traceback (most recent call last):
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/httpserver.py", line 1068, in process_request_in_thread
ckan-dev_1    |     self.finish_request(request, client_address)
ckan-dev_1    |   File "/usr/lib/python2.7/SocketServer.py", line 331, in finish_request
ckan-dev_1    |     self.RequestHandlerClass(request, client_address, self)
ckan-dev_1    |   File "/usr/lib/python2.7/SocketServer.py", line 654, in __init__
ckan-dev_1    |     self.finish()
ckan-dev_1    |   File "/usr/lib/python2.7/SocketServer.py", line 713, in finish
ckan-dev_1    |     self.wfile.close()
ckan-dev_1    |   File "/usr/lib/python2.7/socket.py", line 283, in close
ckan-dev_1    |     self.flush()
ckan-dev_1    |   File "/usr/lib/python2.7/socket.py", line 307, in flush
ckan-dev_1    |     self._sock.sendall(view[write_offset:write_offset+buffer_size])
ckan-dev_1    | error: [Errno 32] Broken pipe
ckan-dev_1    | 2019-05-27 12:56:19,179 INFO  [provider_logger] vito         | 2019-05-27 12:56:06.613714 | 200 | 4.374127s
ckan-dev_1    | 2019-05-27 12:56:22,495 DEBUG [ckanext.nextgeossharvest.harvesters.foodsecurity] next url: https://www.vito-eodata.be/openSearch/findProducts.atom?collection=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR&count=500&start=2017-01-02&end=2019-05-27&startIndex=501
ckan-dev_1    | Traceback (most recent call last):
ckan-dev_1    |   File "/usr/bin/paster", line 11, in <module>
ckan-dev_1    |     load_entry_point('PasteScript==2.0.2', 'console_scripts', 'paster')()
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 102, in run
ckan-dev_1    |     invoke(command, command_name, options, args[1:])
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 141, in invoke
ckan-dev_1    |     exit_code = runner.run(args)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/paste/script/command.py", line 236, in run
ckan-dev_1    |     result = self.command()
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/commands/harvester.py", line 185, in command
ckan-dev_1    |     gather_callback(consumer, method, header, body)
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 298, in gather_callback
ckan-dev_1    |     harvest_object_ids = gather_stage(harvester, job)
ckan-dev_1    |   File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 352, in gather_stage
ckan-dev_1    |     harvest_object_ids = harvester.gather_stage(job)
ckan-dev_1    |   File "/usr/lib/python2.7/site-packages/ckanext/nextgeossharvest/harvesters/foodsecurity.py", line 233, in gather_stage
ckan-dev_1    |     restart_date = open_search_entry.find('dc:date').string.split('/')[1].split('T')[0]  # noqa: E501
ckan-dev_1    | AttributeError: 'NoneType' object has no attribute 'string'

```

joaandrade commented:
Hi @georgiana-b,

Can you send tell us the content of the variable open_search_entry (line before the error) when you run the harvester? We will re-run and check if the content of the open_search_entry variable is the same as yours.

Regards,
João

joaandrade commented:
![debug_food_security](https://user-images.githubusercontent.com/8322111/58423222-79dfd600-808c-11e9-9871-976b5549b5df.png)

Find attached a screenshot of our debug environment. As you see the content of the open_search_entry must cntain the atom for an entry and the tag dc:date must be there.

joaandrade commented:
We've tested right now for the FAPAR collection and the first job collects 1 dataset, second job 81 datasets as it is supposed.
![Screenshot_20190527_144604](https://user-images.githubusercontent.com/8322111/58423954-3dad7500-808e-11e9-83f4-cc41e9b361c7.png)
![Screenshot_20190527_144538](https://user-images.githubusercontent.com/8322111/58423955-3e460b80-808e-11e9-9115-2fbd3f4f2561.png)
![Screenshot_20190527_144244](https://user-images.githubusercontent.com/8322111/58423956-3e460b80-808e-11e9-9570-f0cf4fc0a9d9.png)
![Screenshot_20190527_144229](https://user-images.githubusercontent.com/8322111/58423957-3e460b80-808e-11e9-9f5b-c7963b3f1c41.png)

georgiana-b commented:
Hey @joaandrade! I looked into this quite extensibvely yesterday and I think I finally figured out the issue. So this is the raw value of the `open_search_entry` variable:

```xml
<atom:entry>
<atom:title>urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR:S2A_20170215T105607Z_29PRR_FAPAR_10M_V001</atom:title>
<atom:id>urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR:S2A_20170215T105607Z_29PRR_FAPAR_10M_V001</atom:id>
<dc:identifier>urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR:S2A_20170215T105607Z_29PRR_FAPAR_10M_V001</dc:identifier>
<atom:published>2018-05-09T20:00:56Z</atom:published>
<atom:updated>2018-05-09T20:00:56Z</atom:updated>
<atom:summary>Product belonging to collection Sentinel-2 NEXTGEOSS FAPAR.  It was ingested on the 2018-05-09T20:00:56Z.</atom:summary>
<dc:date>2017-02-15T10:56:07Z/2017-02-15T11:08:37Z</dc:date>
<georss:box>13.464738 -6.229434 14.442159 -5.200492</georss:box>
<georss:polygon>13.464738 -6.2294345 13.464738 -5.200492 14.442159 -5.200492 14.442159 -6.2294345 13.464738 -6.2294345</georss:polygon>
<owc:offering code="http://www.opengis.net/spec/owc-atom/1.0/req/wmts">
<owc:operation code="GetCapabilities" href="http://proba-v-mep.esa.int/applications/geo-viewer/app/mapcache/wmts?service=WMTS&amp;request=GetCapabilities" method="GET" type="application/xml"/>
</owc:offering>
<atom:link href="https://www.vito-eodata.be/cgi-bin/probav?SERVICE=WMS&amp;VERSION=1.3.0&amp;REQUEST=GetMap&amp;SRS=EPSG:4326&amp;FORMAT=image/png&amp;LAYERS=NEXTGEOSS_SENTINEL2_FAPAR_FAPAR&amp;TIME=2017-02-15T10:56:07Z&amp;IDENTITYID=47774644&amp;BBOX=13.464738,-6.229434,14.442159,-5.200492&amp;HEIGHT=200&amp;WIDTH=211" rel="icon" title="FAPAR" type="image/png"/>
<atom:link href="https://www.vito-eodata.be/PDF/dataaccessMdXML??mdmode=Inspire&amp;collectionID=1000144&amp;productID=47774644&amp;fileName=inspire.xml" rel="alternate" title="Inspire" type="application/vnd.iso.19139+xml"/>
<atom:link href="https://www.vito-eodata.be/PDF/dataaccess?service=DSEO&amp;request=GetProduct&amp;version=1.0.0&amp;collectionID=1000144&amp;productID=47774644&amp;ProductURI=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR:S2A_20170215T105607Z_29PRR_FAPAR_10M_V001&amp;" rel="enclosure" title="S2A_20170215T105607Z_29PRR_FAPAR_10M_V001" type="application/octet-stream"/>
<atom:link href="file:///data/MTDA/NEXTGEOSS_SENTINEL2_FAPAR/S2A_20170215T105607Z_29PRR_FAPAR_V001/" rel="enclosure" title="Download" type="text/directory"/>
<atom:link href="https://www.vito-eodata.be/ordering/webservice?wsdl" rel="enclosure" title="HMA ordering" type="application/vnd.ogc.oseo.capabilities.response_xml"/>
<atom:link href="https://www.vito-eodata.be/PDF/dataaccessorder?service=DSEO&amp;request=GetProduct&amp;version=1.0.0&amp;collectionID=1000144&amp;productID=47774644&amp;ProductURI=urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR:S2A_20170215T105607Z_29PRR_FAPAR_10M_V001&amp;" rel="enclosure" title="HTML ordering" type="text/html"/>
<atom:link href="https://www.vito-eodata.be/collections/srv/eng/main.home?uuid=urn:eop:VITO:CGS_S2_FAPAR" rel="describedBy" type="text/html"/>
<atom:content>Referenced as S2A_20170215T105607Z_29PRR_FAPAR_10M_V001</atom:content>
<media:group>
<media:content medium="image" type="image/png" url="https://www.vito-eodata.be/cgi-bin/probav?SERVICE=WMS&amp;VERSION=1.3.0&amp;REQUEST=GetMap&amp;SRS=EPSG:4326&amp;FORMAT=image/png&amp;LAYERS=NEXTGEOSS_SENTINEL2_FAPAR_FAPAR&amp;TIME=2017-02-15T10:56:07Z&amp;IDENTITYID=47774644&amp;BBOX=13.464738,-6.229434,14.442159,-5.200492&amp;HEIGHT=200&amp;WIDTH=211">
<media:title>FAPAR</media:title>
<media:category scheme="http://www.opengis.net/spec/EOMPOM/1.1">QUICKLOOK</media:category>
</media:content>
</media:group>
<EarthObservation gml:id="eopr_47774644" schemaLocation="http://www.opengis.net/eop/2.1 http://schemas.opengis.net/eompom/1.1/xsd/eop.xsd http://www.opengis.net/opt/2.1 http://schemas.opengis.net/eompom/1.1/xsd/opt.xsd http://www.opengis.net/om/2.0 http://schemas.opengis.net/om/2.0/observation.xsd">
<phenomenonTime>
<gml:TimePeriod gml:id="tp_47774644">
<gml:beginPosition>2017-02-15T10:56:07Z</gml:beginPosition>
<gml:endPosition>2017-02-15T11:08:37Z</gml:endPosition>
</gml:TimePeriod>
</phenomenonTime>
<resultTime>
<gml:TimeInstant gml:id="ti_47774644">
<gml:timePosition>2019-05-27T14:56:29.27Z</gml:timePosition>
</gml:TimeInstant>
</resultTime>
<procedure>
<EarthObservationEquipment gml:id="eoe_47774644">
<platform>
<Platform>
<shortName>SENTINEL-2</shortName>
</Platform>
</platform>
<instrument>
<Instrument>
<shortName>MSI</shortName>
</Instrument>
</instrument>
<sensor>
<Sensor>
<sensorType>OPTICAL</sensorType>
</Sensor>
</sensor>
<acquisitionParameters>
<Acquisition>
<orbitNumber>0</orbitNumber>
<orbitDirection/>
</Acquisition>
</acquisitionParameters>
</EarthObservationEquipment>
</procedure>
<observedProperty href="#params1"/>
<featureOfInterest>
<Footprint gml:id="fp_47774644">
<multiExtentOf>
<gml:MultiSurface gml:id="ms_47774644" srsName="EPSG:4326">
<gml:surfaceMembers>
<gml:Polygon gml:id="fppoly_47774644">
<gml:exterior>
<gml:LinearRing>
<gml:posList>13.464738 -6.2294345 13.464738 -5.200492 14.442159 -5.200492 14.442159 -6.2294345 13.464738 -6.2294345</gml:posList>
</gml:LinearRing>
</gml:exterior>
</gml:Polygon>
</gml:surfaceMembers>
</gml:MultiSurface>
</multiExtentOf>
<centerOf>
<gml:Point gml:id="pt_47774644" srsName="EPSG:4326">
<gml:pos>13.953448 -5.714963</gml:pos>
</gml:Point>
</centerOf>
</Footprint>
</featureOfInterest>
<result>
<EarthObservationResult gml:id="optr_47774644">
<browse>
<BrowseInformation>
<type>QUICKLOOK</type>
<referenceSystemIdentifier codeSpace="EPSG">epsg:4326</referenceSystemIdentifier>
<fileName>
<ServiceReference href="https://www.vito-eodata.be/cgi-bin/probav?SERVICE=WMS&amp;VERSION=1.3.0&amp;REQUEST=GetMap&amp;SRS=EPSG:4326&amp;FORMAT=image/png&amp;LAYERS=NEXTGEOSS_SENTINEL2_FAPAR_FAPAR&amp;TIME=2017-02-15T10:56:07Z&amp;IDENTITYID=47774644&amp;BBOX=13.464738,-6.229434,14.442159,-5.200492&amp;HEIGHT=200&amp;WIDTH=211">
<RequestMessage/>
</ServiceReference>
</fileName>
</BrowseInformation>
</browse>
<product>
<ProductInformation>
<fileName>
<ServiceReference href="https://www.vito-eodata.be/PDF/dataaccess?service=DSEO&amp;request=GetProduct&amp;version=1.0.0&amp;collectionID=1000144&amp;productID=47774644&amp;ProductURI=47774644">
<RequestMessage/>
</ServiceReference>
</fileName>
<size uom="Bytes">7208260</size></ProductInformation>
</product>
<cloudCoverPercentage uom="%">81.128</cloudCoverPercentage>
</EarthObservationResult>
</result>
<metaDataProperty>
<EarthObservationMetaData>
<identifier>urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR:S2A_20170215T105607Z_29PRR_FAPAR_10M_V001</identifier>
<parentIdentifier>urn:eop:VITO:NEXTGEOSS_SENTINEL2_FAPAR</parentIdentifier>
<acquisitionType>NOMINAL</acquisitionType>
<productType>NEXTGEOSS_SENTINEL2_FAPAR</productType>
<status>ARCHIVED</status>
<vendorSpecific>
<SpecificInformation>
<localAttribute>ProductReference</localAttribute>
<localValue>S2A_20170215T105607Z_29PRR_FAPAR_10M_V001</localValue>
</SpecificInformation>
</vendorSpecific>
<vendorSpecific>
<SpecificInformation>
<localAttribute>ProductVersion</localAttribute>
<localValue>001</localValue>
</SpecificInformation>
</vendorSpecific>
<vendorSpecific>
<SpecificInformation>
<localAttribute>MissingDataPercentage</localAttribute>
<localValue>92.004</localValue>
</SpecificInformation>
</vendorSpecific>
<vendorSpecific>
<SpecificInformation>
<localAttribute>MapProjectionReference</localAttribute>
<localValue>EPSG:32629</localValue>
</SpecificInformation>
</vendorSpecific>
</EarthObservationMetaData>
</metaDataProperty>
</EarthObservation><!-- OM11 timestamp is 2052 ms. -->
</atom:entry>
```

But running `open_search_entry.find('dc:date')`  was returning `None` even though the tag is there.  Eventually I realized that when the xml is parsed by beautifulsoup, the prefix of the tag is removed. Here you can see  how the `date` tag changes in the beautifulsoup object as opposed to the original:

![image](https://user-images.githubusercontent.com/8862002/58462470-45265a00-8132-11e9-91d1-6d9fc88f38b1.png)

So I just changed [line 233](https://github.com/NextGeoss/ckanext-nextgeossharvest/blob/053c0d388b1917516a0f5adbee70971a98e4ea9d/ckanext/nextgeossharvest/harvesters/foodsecurity.py#L233) to search for `date` instead of `dc:date`:

```py
restart_date = open_search_entry.find('date').string.split('/')[1].split('T')[0]
```

After the change I was also able to import the datasets.
I wonder why you didn't encounter this issue. Maybe we use a different version of BeautifulSoup? Mine is `beautifulsoup 4.5.1`.

Alternatively you could just make the change I made and see if you can retrieve the datasets. In that case it would be best to go with the version that works for both of us.

joaandrade commented:
Hi @georgiana-b,

I guess it should be related with the version of python-bs4 yes. Mine is 4.4.1, so you have the latest one.
I've tested on my environment using only "date" instead of "dc:date" and it still works. I will update the code and it should work on both environments.

Great job!

joaandrade commented:
Hi again @georgiana-b 

I've figured out that there is another line of the code using the "find" with dc:date (line 260 within the function _get_last_harvesting_date). Can you put a breakpoint on this line and check the behavior on your side? Because on my code this line only works with "dc:date".

It would be important to understand what happens in your code on this line to check if we also have to update it.

georgiana-b commented:
@joaandrade Perhaps a better solution would ve to ceil the version of beautifulsoup needed in the project by specifying the version that worked for you in [the requirements file](https://github.com/NextGeoss/ckanext-nextgeossharvest/blob/053c0d388b1917516a0f5adbee70971a98e4ea9d/requirements.txt).

joaandrade commented:
@georgiana-b we can do that too. In any case can you check the behavior of the line 260 on your environment?

georgiana-b commented:
Sure, I'm on it.

joaandrade commented:
Just an additional information. The line 260 runs first, since it is here that is defined the restart date for the next job. So you probably will be able to test it after colllect the first dataset (first job).

georgiana-b commented:
Ok.
For the first test, I get the same error on line 233 even with `beautifulsoup 4.4.1` which is quite peculiar.

joaandrade commented:
It is weird in fact. But for the line 233 I'not concerned since my version works properly removing the "dc:". The main problem is on the line 260. On this line, if I remove the "dc:" I got the error "nonetype has no string..". My guess is that on the line 233 the open_search_entry is not a soup anymore. However, on the line 260 it is a soup for sure.

You can update the line 233 to "date" (it works for me too). Then on the second job test the line 260.

georgiana-b commented:
@joaandrade Interestingly enough, I encounter no issues on line 260 as it is now. 
The `restart_date` [there](https://github.com/NextGeoss/ckanext-nextgeossharvest/blob/053c0d388b1917516a0f5adbee70971a98e4ea9d/ckanext/nextgeossharvest/harvesters/foodsecurity.py#L259) has the right value, no errors.

So I guess the solution is to only change line 233 with `find('date')` and leave the other as it is.

joaandrade commented:
I agree. The update was performed and commited. I guess the justification is the one on my previous comment. Let me know if you detect any additional issue.

georgiana-b commented:
Glad we figured it out. Thank you for your patience!

joaandrade commented:
Hi @georgiana-b ,

I just figured out thet on your comment from 2 days ago, you've identified 2 errors. the one with the opensearch entry was already solved but the second one seems to be the same problem you are experiencing on David's harvester.
I checked on my SOLR the spatial field for the FAPAR datasets and they have the commas

![image](https://user-images.githubusercontent.com/8322111/58566173-05419e80-8228-11e9-83dc-6cf75eeb6f2a.png)

Maybe theis problem you are experiencing on both harvester is related to your ckanext-spatial or solr. Have you tried to restart jetty?

Hope this helps.
João

georgiana-b commented:
Hi @joaandrade! Good catch!

At this point I'm quite confident there is an issue in our setup. I'm currently looking into it.
Knowing this occurred in other harvesters provides a valuable clue. Thanks a lot for pointing that out!

georgiana-b commented:
@JGulic This harvester was tested this quite extensively, including on staging, and there were no other issues so it's good to go.
